### DISCUSSION TITLE
kohya SS gui optimal parameters - Kohya DyLoRA , Kohya LoCon , LyCORIS/LoCon , LyCORIS/LoHa , Standard

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [kohya-ss/sd-scripts#441] [Q&A]
I am working on a new tutorial to hopefully publish on my channel : https://www.youtube.com/secourses

I got first training done but there are so many parameters and which are most optimal i wonder

These were the settings I have used

So If i can learn better i can explain people better

`
accelerate launch --num_cpu_threads_per_process=16 "train_network.py" --enable_bucket --pretrained_model_name_or_path="F:/0 models/v1-5-pruned.ckpt" --train_data_dir="C:\Users\King\Pictures\kohya\img" --reg_data_dir="C:\Users\King\Pictures\kohya\reg" --resolution=512,512 --output_dir="C:\Users\King\Pictures\kohya\model" --logging_dir="C:\Users\King\Pictures\kohya\log" --network_alpha="8" --save_model_as=safetensors --network_module=lycoris.kohya --network_args "conv_dim=8" "conv_alpha=1" "algo=lora" --text_encoder_lr=5e-5 --unet_lr=0.0001 --network_dim=16 --output_name="test1" --lr_scheduler_num_cycles="200" --learning_rate="0.0001" --lr_scheduler="cosine" --lr_warmup_steps="340" --train_batch_size="1" --max_train_steps="3400" --save_every_n_epochs="10" --mixed_precision="bf16" --save_precision="bf16" --caption_extension=".txt" --cache_latents --optimizer_type="AdamW8bit" --max_data_loader_n_workers="0" --bucket_reso_steps=64 --xformers --bucket_no_upscale --sample_sampler=euler_a --sample_prompts="C:\Users\King\Pictures\kohya\model\sample\prompt.txt" --sample_every_n_epochs="5"`

so which one best?

Kohya DyLoRA , Kohya LoCon , LyCORIS/LoCon , LyCORIS/LoHa , Standard

Also which learning rates and Network Rank (Dimension) , Network Alpha i need for each one?

So far i find that LyCORIS/LoCon performs better than default Standard with above settings 

@kohya-ss 

if you could guide me i could explain every parameters in the video

I am using web ui version

![image](https://user-images.githubusercontent.com/19240467/233994299-a9de4909-c7fa-474d-8a33-69d551661a1a.png)

**Comment by kuzman123** [comment]
I experimented a bit with LoCon and loha and the conclusions are as follows:
1. I'v trained LoCon but with a very specific data set, which consists of 3 different subjects (trained together with captions, and train data set was in one folder) and I wanted to morph them (close up of machines, samurai and cyberpunk ppl). The same I also did with the standard Lora. LoCon did a much better job of merging the subjects, it morphed them into a single subject perfectly and it's very flexibile, while I got much worse results with the standard Lora (the model is not unusable, but it doesn't have that crystal definition like LoCon).
2. LoHa does the opposite of LoCon and Lora, and it's main feature is to prevent "bleeding" between concepts, but in cases where you have separate data sets (which is supposedly what both LoCon and Lora are capable of doing, but they bleed a bit, at least that's what the expert says, whose text I read and saw examples of). With LoHa, you can have completely separate concepts, styles, subjects, etc. I didn't get to try it out, but I've seen examples and it works really well.
![01](https://github.com/kohya-ss/sd-scripts/assets/134649217/fb15382d-e4df-4880-91d1-a36d49c7a151)
![00010-5764465](https://github.com/kohya-ss/sd-scripts/assets/134649217/81cea019-5110-4423-a7fc-3c8592640b35)
